File: SPRINT_2/Output/SDK/Samples/NET/CS WPF Pump Sample/SaleItem.cs
using System;
using System.Collections.Generic;
using System.Text;

using ITL.Enabler.API;

namespace ITL.Enabler.Samples
{
    /// <summary>
    /// Type of item in sale
    /// </summary>
    public enum SalesItemType { NoItem = 0, DeliveryItem = 1, prepayItem = 2 };

 
    /// <summary>
    /// Class contains a single sales item which is saved in a sale.
    /// Encapsulates all details of sale item.
    /// </summary>
    public class SaleItem
    {
         /// <summary>
        /// Type of sales item
        /// </summary>
        public SalesItemType ItemType { get; set; }

        /// <summary>
        /// Item description
        /// </summary>
        public string Description { get; set; }
 
        /// <summary>
        /// Price of item
        /// </summary>
        public decimal Price { get; set; }

         /// <summary>
        /// Return formatted string of price with current currency
        /// </summary>
        public string FormattedPrice
        {
            get
            {
                return String.Format("{0:c2}", Price);
            }
        }

        /// <summary>
        /// Quantity or volume
        /// </summary>
        public decimal Quantity { get; set; }

        public string FormattedQuantity
        {
            get
            {
                if ( ItemType == SalesItemType.NoItem )
                    return String.Format("{0:f02}", Quantity );
                else
                    return String.Format("{0:f03}L", Quantity);
            }
        }

        /// <summary>
        /// Total value of item
        /// </summary>
        public decimal Value { get; set; }

        /// <summary>
        /// Return formatted string of value with current currency
        /// </summary>
        public string FormattedValue
        {
            get
            {
                return String.Format("{0:c2}", Value );
            }

        }
        
        /// <summary>
        /// Constructor for sale item
        /// </summary>
        /// <param name="itemType">Type of item</param>
        /// <param name="trans">Associatted fuel transaction or null</param>
        /// <param name="description">Item description</param>
        /// <param name="price">Price</param>
        /// <param name="quantity">Quantity or volume</param>
        /// <param name="value">Total value of item</param>
        public SaleItem(SalesItemType itemType, string description, decimal price, decimal quantity, decimal value)
        {
            ItemType = itemType;
            Description = description;
            Price = price;
            Quantity = quantity;
            Value = value;
        }
    }


    public class PrepayFuelItem : SaleItem
    {
        public PrepayFuelItem(string description, decimal price, decimal quantity, decimal value, Transaction trans, PumpAuthoriseLimits authData)
            : base(SalesItemType.prepayItem, description, price, quantity, value)
        {
            AuthData = authData;
            Trans = trans;
        }

        /// <summary>
        /// Authorise limits for pump
        /// </summary>
        public PumpAuthoriseLimits AuthData { get; set; }

        /// <summary>
        /// Related fuel transaction or null
        /// </summary>
        public Transaction Trans { get; set; } // Related fuel transaction

    }

    public class PostPayFuelItem : SaleItem
    {
        public PostPayFuelItem(string description, decimal price, decimal quantity, decimal value, Transaction trans)
            : base(SalesItemType.DeliveryItem, description, price, quantity, value)
        {
            Trans = trans;
        }
         /// <summary>
        /// Related fuel transaction or null
        /// </summary>
        public Transaction Trans { get; set; } // Related fuel transaction
    }
}

File: FINAL - same as root/Output/SDK/Samples/Java/pumpdemo/src/pumpdemo/sale/SaleItem.java
/**
 * SaleItem.java
 * 
 * Copyright 1997-2012 Integration Technologies Limited
 * All rights reserved.
 * 
 */
package pumpdemo.sale;

import java.text.NumberFormat;

/**
 * Contains a single sales item
 */
public class SaleItem {

	/**
	 * @return the {@link SaleItemType}
	 */
	public SaleItemType getSaleItemType() {
		return itemType;
	}

	/**
	 * @return the item description
	 */
	public String getDescription() {
		return description;
	}

	/**
	 * @return the price of the item
	 */
	public double getPrice() {
		return price;
	}

	/**
	 * @return the quantity or volume of the item
	 */
	public double getQuantity() {
		return quantity;
	}

	/**
	 * @return the total value of the item
	 */
	public double getValue() {
		return value;
	}

	/**
	 * @param itemType
	 *            the {@link SaleItemType} to set
	 */
	public void setSaleItemType(SaleItemType saleItemType) {
		this.itemType = saleItemType;
	}

	/**
	 * @param description
	 *            the item description to set
	 */
	public void setDescription(String description) {
		this.description = description;
	}

	/**
	 * @param price
	 *            the item price to set
	 */
	public void setPrice(double price) {
		this.price = price;
	}

	/**
	 * @param quantity
	 *            the quantity or volume of the item to set
	 */
	public void setQuantity(double quantity) {
		this.quantity = quantity;
	}

	/**
	 * @param value
	 *            the value of the item to set
	 */
	public void setValue(double value) {
		this.value = value;
	}

	/**
	 * @return the price formatted as a String with the current currency
	 */
	public String getFormattedPrice() {

		return NumberFormat.getCurrencyInstance().format(price);
	}

	/**
	 * @return the value formatted as a String with the current currency
	 */
	public String getFormattedValue() {

		return NumberFormat.getCurrencyInstance().format(value);
	}

	/**
	 * @return the quantity formatted as a String
	 */
	public String getFormattedQuantity() {

		// Can be made locale sensitive
		if (itemType == SaleItemType.NO_ITEM) {
			return String.format("%.2f", quantity);
		} else {
			return String.format("%.2fL", quantity);
		}
	}

	/**
	 * @param itemType
	 *            {@link SaleItemType}
	 * @param description
	 *            Item description
	 * @param price
	 *            Price of sale item
	 * @param quantity
	 *            Quantity of volume of sale item
	 * @param value
	 *            Total value of sale item
	 */
	public SaleItem(SaleItemType saleItemType, String description,
			double price, double quantity, double value) {
		this.itemType = saleItemType;
		this.description = description;
		this.price = price;
		this.quantity = quantity;
		this.value = value;
	}

	private SaleItemType itemType;
	private String description;

	private double price;
	private double quantity;
	private double value;

}
